packages/pangea-sdk/src/main/java/cloud/pangeacyber/pangea/intel/DomainIntelClient.java

```diff
@@ -119,13 +150,99 @@ public DomainLookupResponse lookup(String domain, boolean verbose, boolean raw)
      * @pangea.code
      * {@code
      * DomainLookupResponse response = client.lookup(
-     *     "737updatesboeing.com", 
-     *     "domaintools", 
-     *     true, 
+     *     "737updatesboeing.com",
+     *     "domaintools",
+     *     true,
      *     true);
      * }
      */
     public DomainLookupResponse lookup(String domain, String provider, boolean verbose, boolean raw) throws PangeaException, PangeaAPIException {
         return lookupPost(domain, provider, verbose, raw);
     }
+
+    private DomainReputationResponse reputationPost(String domain, String provider, Boolean verbose, Boolean raw) throws PangeaException, PangeaAPIException{
+        DomainReputationRequest request = new DomainReputationRequest(domain, provider, verbose, raw);
+        DomainReputationResponse resp = doPost("/v1/reputation", request, DomainReputationResponse.class);
+        return resp;
+    }
+
+    /**
+     * Look up a domain reputation
+     * @pangea.description Retrieve domain reputation from a default provider.
+     * @param domain domain address to be looked up
+     * @return DomainReputationResponse
+     * @throws PangeaException
+     * @throws PangeaAPIException
+     * @pangea.code
+     * {@code
+     * DomainReputationResponse response = client.reputation(
+     *     "737updatesboeing.com");
+     * }
+     */
+    public DomainReputationResponse reputation(String domain) throws PangeaException, PangeaAPIException {
+        return reputationPost(domain, null, null, null);
+    }
+
+    /**
+     * Look up a domain - domain, provider
+     * @pangea.description Retrieve domain reputation for a particular provider
+     * @param domain domain address to be looked up
+     * @param provider provider to get reputation from
+     * @return DomainReputationResponse
+     * @throws PangeaException
+     * @throws PangeaAPIException
+     * @pangea.code
+     * {@code
+     * DomainReputationResponse response = client.reputation(
+     *     "737updatesboeing.com",
+     *     "domaintools");
+     * }
+     */
+    public DomainReputationResponse reputation(String domain, String provider) throws PangeaException, PangeaAPIException {
```

Comment:
does it make sense to have these overloaded calls, we would need every permutation of the raw, provider and verbose parameters since you can give any combination of those. why not just give the full set of parameters?

Reply:
I think we couldn't have overloads for any combination because raw and verbose are both bool. I've choose this 4 overloads because I considerate they cover so many cases and avoid the user to use so many parameters on null, which would be awful.